rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -403,30 +406,64 @@ impl TunnelManager {
     }
 
     /// Given a LocalIdentity, connect to the neighbor over wireguard
+    /// return the tunnel object and if already had a tunnel
     pub fn open_tunnel(
         &mut self,
         their_localid: LocalIdentity,
         peer: Peer,
         our_port: u16,
-    ) -> Result<Tunnel, Error> {
-        trace!("TunnelManager getting existing tunnel or opening a new one");
-        // This is deceptively simple, see the commit message
+    ) -> Result<(Tunnel, bool), Error> {
+        trace!("getting existing tunnel or opening a new one");
+        // ifidx must be a part of the key so that we can open multiple tunnels
+        // if we have more than one physical connection to the same peer
         let key = &(their_localid.global.clone(), peer.ifidx);
-        if self.tunnels.contains_key(key) {
+
+        let we_have_tunnel = self.tunnels.contains_key(key);
+        let they_have_tunnel = match their_localid.have_tunnel {
+            Some(v) => v,
+            None => true, // when we don't take the more conservative option
```

Comment:
This reads like its wrong: `they_have_tunnel` is `true` when `their_lcoalid.have_tunnel` is `None`?

Reply:
it's an assumption, we don't know if they have a tunnel so we take the more conservative option and assume they do.